app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,19 @@
+class CommentsController < ApplicationController
+  def new; end
+
+  def create
+    @current_user = current_user
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @comment = @current_user.comments.new(comments_params)
+    @comment.post = @post
+    @comment.save
+    redirect_to user_post_path(@user, @post)
+  end
```

Comment:
- [ ] <b>REQUIRED:</b> There are few complications in this method, one user on line 5 as `@current user` amd another user on line 6 as `@user`, so which one should be used in this case?  And also, you did not handle the exception in case the comment failed to be created. Kindly address those issues. Here is the alternative on how you can fix those issues.

```ruby
 def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      redirect_to post_path(@post)
    end
  end
```